core/src/main/java/org/pentaho/platform/util/StringUtil.java

```diff
@@ -116,4 +118,31 @@ public static String getMapAsPrettyString( final Map map ) {
     MapUtils.debugPrint( new PrintStream( writtenBytes ), "Map", map );
     return writtenBytes.toString();
   }
+
+  /**
+   * Returns a map of String key-value pairs, in which each key and corresponding value has been trimmed of whitespace.
+   * @param map the {@link Map} of String key-value pairs to be trimmed
+   * @return a trimmed String {@link Map}
+   */
+  public static Map<String, String> trimStringMap( Map<String, String> map ) {
```

Comment:
trim might be a better name, we know we're trimming a map because of the param type - but that's just my opinion, no need to take action if you disagree